gapis/api/vulkan/frame_loop.go

```diff
@@ -404,7 +428,57 @@ func (f *frameLoop) buildStartEndStates(ctx context.Context, startState *api.Glo
 				f.descriptorSetLayoutDestroyed[descriptorSetLayout] = true
 			}
 
-			// TODO: Recreate destroyed resources.
+		// Semaphores
+		case *VkCreateSemaphore:
+			vkCmd := cmd.(*VkCreateSemaphore)
+			sem := vkCmd.PSemaphore().MustRead(ctx, vkCmd, currentState, nil)
+			log.D(ctx, "Semaphore %v is created during loop.", sem)
+			f.semaphoreCreated[sem] = true
+
+		case *VkDestroySemaphore:
+			vkCmd := cmd.(*VkDestroySemaphore)
+			sem := vkCmd.Semaphore()
+			log.D(ctx, "Semaphore %v is destroyed during loop.", sem)
+			if _, ok := f.semaphoreCreated[sem]; !ok {
+				f.semaphoreDestroyed[sem] = true
+			} else {
+				delete(f.semaphoreCreated, sem)
```

Comment:
should all these deletes also delete the mapping from f.<type>Changed?

Reply:
Thanks Pascal Muetschard for helping review the code. 

Yes you're right,need to exclude the f.change*. However the f.changed* is only available at the end of the simulation loop. So updated the code at the reset part to filter out the create/destroy for changed resource in the resetting step.